packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -432,19 +432,30 @@ export class FluidDataStoreRuntime
 		return context.getChannel();
 	}
 
-	public createChannel(id: string = uuid(), type: string): IChannel {
+	public createChannel(channel: IChannel): void;
+	public createChannel(id: string | undefined, type: string): IChannel;
+	createChannel(idOrChannel: string | IChannel = uuid(), maybeType?: string): IChannel | void {
+		const id = typeof idOrChannel === "string" ? idOrChannel : idOrChannel.id;
 		if (id.includes("/")) {
 			throw new UsageError(`Id cannot contain slashes: ${id}`);
 		}
 
 		this.verifyNotClosed();
 
 		assert(!this.contexts.has(id), 0x179 /* "createChannel() with existing ID" */);
+
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const type = typeof idOrChannel === "string" ? maybeType! : idOrChannel.attributes.type;
+		const factory = this.sharedObjectRegistry.get(type);
```

Comment:
If we keep this assertion (I'm with Matt and Tyler [here](https://github.com/microsoft/FluidFramework/pull/18418/files#r1400903702), which would probably make it go away), I think we should write _why_ it is ok to use in this particular instance. IMO every single lint-disable that we write should have an explicit justification.

That said, we're losing some validation here and I'd say it's a bug. `type` used to be `assert`-ed to not be undefined, but in this version of the code someone could still technically pass a string id and an undefined type, so using a non-null assertion feels incorrect to me.

Reply:
done